### PR TITLE
Add script to validate the Berlin mask

### PR DESCRIPTION
**Pull request by hsonne**
I added a script to validate the Berlin mask provided by @amatzi. In the script, I compare his matrix with a matrix that I obtain by applying `raster::mask()` to the full German raster with a Berlin shape obtained from https://gadm.org/

I found slight differences (red: set only in my mask from gadm's shape file, green: set only in mask by @amatzi, blue: set in both masks):

grafik

I assume it has to do with projection.

@LukasGue You may want to check the effect of these differences by comparing the results of both versions of the `load_potential_evaporation_berlin()` function.

I found e.g.
```
>   # Compare the results of the two versions
>   result_1
                                         file year month      mean        sd   min   max n_values
1 grids_germany_monthly_evapo_p_202106.asc.gz 2021     6 141.06251 0.8375917 138.3 143.3     1019
2 grids_germany_monthly_evapo_p_202107.asc.gz 2021     7 120.99951 0.8580314 118.3 123.1     1019
3 grids_germany_monthly_evapo_p_202108.asc.gz 2021     8  86.29107 0.5446369  84.5  87.4     1019
>   result_2
                                         file year month      mean        sd   min   max n_values
1 grids_germany_monthly_evapo_p_202106.asc.gz 2021     6 141.11486 0.8081033 138.3 143.3      902
2 grids_germany_monthly_evapo_p_202107.asc.gz 2021     7 121.06552 0.8237308 118.3 123.1      902
3 grids_germany_monthly_evapo_p_202108.asc.gz 2021     8  86.33326 0.5241792  84.5  87.6      902
```

by running this part of the script:
https://github.com/KWB-R/kwb.dwd/blob/190c2d6c47994064b9a5e58cd90b044e5f77deaf/inst/extdata/validate-berlin-mask.R#L15

---
This change is [Reviewable](https://reviewable.io/reviews/kwb-r/kwb.dwd/29)

### CODE REVIEW

**Comment by LukasGue** [comment]
The comparison of the two methods resulted in a **MAPE of 0.08095465 %** for 30 years of annual mean potential evaporation. In my opinion, this slight difference is neglectable.

File: R/check_shapes_germany.R [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#' @importFrom kwb.utils createDirectory removeExtension
+check_shapes_germany <- function()
+{
+  url <- "https://geodata.ucdavis.edu/gadm/gadm4.0/shp/gadm40_DEU_shp.zip"
+
+  zip_name <- basename(url)
+  shape_name <- kwb.utils::removeExtension(zip_name)
+  shape_dir <- temp_dir(shape_name)
+
+  if (! dir.exists(shape_dir)) {
```
Comment (LukasGue) [inline R/check_shapes_germany.R:12]: Some comments here might help

File: R/load_monthly_variable_for_region.R [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+)
+{
+  # Currently, two variables are supported
+  match.arg(variable, c("precipitation", "evapo_p"))
+
+  # Get URLs to .asc.gz files with monthly grids on DWD server
+  base_url <- ftp_path_monthly_grids(variable)
+  urls <- list_zipped_esri_ascii_grids(base_url, from, to)
+
+  if (version == 1L) {
```
Comment (LukasGue) [inline R/load_monthly_variable_for_region.R:13]: What is this version 1 or 2?

File: R/check_shapes_germany.R (hunk 1 above)
Comment (hsonne) [reply R/check_shapes_germany.R:26]: Like so? 0a1eac8e6509e20f026d6b2529ab111039abee4f

File: R/select_shapes.R [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+{
+  choices <- names(shapes)
+
+  selection <- utils::select.list(choices, title = "Select layer")
+
+  which(selection == choices)
+}
+
+# select_variable --------------------------------------------------------------
+select_variable <- function(shape)
```
Comment (LukasGue) [inline R/select_shapes.R:36]: Not quite sure, which variable is meant here. Maybe a few comments in these lines can help understand.

File: R/check_shapes_germany.R (hunk 1 above)
Comment (LukasGue) [reply R/check_shapes_germany.R:26]: Yes, the code is quite readable, but the comments help reading the code and avoid misunderstanding.

File: R/load_monthly_variable_for_region.R (hunk 2 above)
Comment (hsonne) [reply R/load_monthly_variable_for_region.R:13]: Not exactly. 

Version 1 is the old version, 
* reading "manually" the matrix values from the `.asc` file,
* masking them "manually" by multiplying with a matrix that was "manually" created by Andreas for Berlin.

Version 2 is the new version, 
* using `raster::raster()` to read the .asc file and also providing the coordinate reference system by placing a .prj file of same name
* using `raster::mask()` with a shape for a German region

**Review by LukasGue — APPROVED** [review]
All scripts reviewed, some tested in R. Some comments already given in respective lines.

**Comment by hsonne** [comment]
@LukasGue Thanks for reviewing!

**Pull request merged**